src/main/Main.hs

```diff
@@ -848,6 +852,16 @@ execCmd ExecOpts {..} go@GlobalOpts{..} = do
                munlockFile lk -- Unlock before transferring control away.
                exec eoEnvSettings cmd args
 
+-- | Evaluate some haskell code inline.
+evalCmd :: ExecOpts -> GlobalOpts -> IO ()
+evalCmd e@ExecOpts { eoArgs = args } go@GlobalOpts {..} =
+    if length args /= 1
+    then error "Expecting ONE string argument to 'eval'"
```

Comment:
I was trying to reuse the `execOptsParser`, which should not throw an error when given multiple arguments. 

I can create a separate parser for this, but it would have to duplicate some code from `execOptsParser`. As I expect `stack eval CODE` to behave exactly like `stack exec ghc -- -e CODE`, this is a fragile solution.